### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by pellared**
## Why

Part of https://github.com/streetsidesoftware/cspell/issues/3270

## What

Add `Dockerfile` for a building container image with `cspell-cli`.

## Testing

CI - added a new `test-docker` job to `Test` GitHub workflow

Linux and macOS (Bash or Zsh):

```sh
docker build -t cspell . && docker run -v $(pwd):/workdir cspell 
```

Windows (Git Bash):

```sh
docker build -t cspell . && docker run -v $(pwd -W):/workdir cspell 
```

Windows (PowerShell):

```powershell
docker build -t cspell .
docker run -v ${PWD}:/workdir cspell 
```

## Follow-ups

Preferably as separate PRs:

1. Add/update release GitHub workflow that publishes the container image to GitHub docker registry. See: https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages
1. Add instructions to `README.md` describing how to use the docker image.
1. Add dependabot configuration to bump the `node` base image (but minor and patch so that we stay on the Node.js LTS version)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -61,3 +61,13 @@ jobs:
...
       # Ensure the repository is clean after build & test
       - run: git --no-pager diff --compact-summary --exit-code
+
+  test-docker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - run: docker build -t cspell .
+      - run: docker run -it -v $(pwd):/workdir cspell
```
Comment (pellared) [inline .github/workflows/test.yml:71]: Do you want it to be moved to `package.json` in `scripts`? If so I would prefer to do it in separate PR as I do not have `node` installed and I have no idea if `$(pwd)` will run out of the box.

**Comment by Jason3S** [comment]
@pellared,

Thank you.
